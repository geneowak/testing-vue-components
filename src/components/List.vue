<template>
  <div>
    <ul>
      <ListItem v-for="(movie, index) in marvelMovies" :key="index" :movie="movie" />
    </ul>
  </div>
</template>

<script>
  import ListItem from './ListItem';

  export default {
    components: {
      ListItem,
    },

    data() {
      return {
        marvelMovies: ['Iron Man', 'Avengers', 'Infitinity War'],
      };
    },
  };
</script>

<style scoped></style>
