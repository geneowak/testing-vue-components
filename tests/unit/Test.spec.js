import TestComponent from '@/components/Test';
import List from '@/components/List';
import { mount } from '@vue/test-utils';

test('mount a vue component', () => {
  const wrapper = mount(TestComponent, {
    propsData: { value: 'World' },
  });
  expect(wrapper).toMatchSnapshot();
});

test('ListComponent', async () => {
  const wrapper = mount(List);
  const movies = wrapper.vm.marvelMovies;

  await wrapper.setData({
    marvelMovies: [...movies, 'Black Widow'],
  });

  expect(wrapper).toMatchSnapshot();
});
