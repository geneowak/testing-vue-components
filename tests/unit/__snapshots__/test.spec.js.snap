// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListComponent 1`] = `
<div>
  <ul>
    <li><strong>Iron Man</strong></li>
    <li><strong>Avengers</strong></li>
    <li><strong>Infitinity War</strong></li>
    <li><strong>Black Widow</strong></li>
  </ul>
</div>
`;

exports[`mount a vue component 1`] = `
<div>
  <h1>Hello World</h1>
</div>
`;
